### ISSUE TITLE
kotlinc doesn't show errors from stderr

### ISSUE THREAD

**Issue by LovecraftianHorror** [dense-analysis/ale#3011]
## Information

**VIM version**

NVIM v0.4.3
Build type: Release

Operating System: Arch Linux, Linux Kernel 5.5.4

## What went wrong

I was trying to use `kotlinc` to lint kotlin files and noticed it wasn't showing any lints, even from the following file `hello.kt`

```kotlin
fun main(args: Array<String>) {
	bad
}
```

Running essentially the same command listed in `:ALEInfo` `/bin/zsh -c 'kotlinc "/home/lovecraft/hello.kt"'` returns

```text
/home/lovecraft/hello.kt:2:2: error: unresolved reference: bad
	bad
 ^

```

I think the problem stems from `kotlinc` outputting to stderr instead of stdout. My hacky way of testing this was to move `kotlinc` to `kotlinc.orig` and create a new `kotlinc` that redirects stderr to stdout like below

```bash
#!/usr/bin/bash

kotlinc.orig "$@" 2>&1
```

After that ALE correctly displays the error in `hello.kt`. Hopefully any of this is helpful, I really appreciate all the work the ALE maintainers have put into such a great plugin!

## Reproducing the bug

1. Install `kotlinc`
2. Create a file with the same contents as `hello.kt`
3. Run ALE and see there are no errors shown

### :ALEInfo

```text
 Current Filetype: kotlin
Available Linters: ['kotlinc', 'ktlint', 'languageserver']
  Enabled Linters: ['kotlinc', 'ktlint', 'languageserver']
 Suggested Fixers: 
  'remove_trailing_lines' - Remove all blank lines at the end of a file.
  'trim_whitespace' - Remove all trailing whitespace characters at the end of every line.
 Linter Variables:

let g:ale_kotlin_kotlinc_classpath = ''
let g:ale_kotlin_kotlinc_config_file = '.ale_kotlinc_config'
let g:ale_kotlin_kotlinc_enable_config = 0
let g:ale_kotlin_kotlinc_module_filename = 'module.xml'
let g:ale_kotlin_kotlinc_options = ''
let g:ale_kotlin_kotlinc_sourcepath = ''
let g:ale_kotlin_kotlinc_use_module_file = 0
let g:ale_kotlin_languageserver_executable = 'kotlin-language-server'
 Global Variables:

let g:ale_cache_executable_check_failures = v:null
let g:ale_change_sign_column_color = 0
let g:ale_command_wrapper = ''
let g:ale_completion_delay = v:null
let g:ale_completion_enabled = 0
let g:ale_completion_max_suggestions = v:null
let g:ale_echo_cursor = 1
let g:ale_echo_msg_error_str = 'E'
let g:ale_echo_msg_format = '[%linter%] [%severity%] %code:%%s'
let g:ale_echo_msg_info_str = 'Info'
let g:ale_echo_msg_warning_str = 'W'
let g:ale_enabled = 1
let g:ale_fix_on_save = 0
let g:ale_fixers = {'c': ['clang-format'], 'rust': ['rustfmt'], 'cpp': ['clang-format'], 'python': ['black']}
let g:ale_history_enabled = 1
let g:ale_history_log_output = 1
let g:ale_keep_list_window_open = 0
let g:ale_lint_delay = 1000
let g:ale_lint_on_enter = 1
let g:ale_lint_on_filetype_changed = 1
let g:ale_lint_on_insert_leave = 1
let g:ale_lint_on_save = 1
let g:ale_lint_on_text_changed = 'normal'
let g:ale_linter_aliases = {}
let g:ale_linters = {}
let g:ale_linters_explicit = 0
let g:ale_list_vertical = 0
let g:ale_list_window_size = 10
let g:ale_loclist_msg_format = '[%linter%] [%severity%] %code:%%s'
let g:ale_lsp_root = {}
let g:ale_max_buffer_history_size = 20
let g:ale_max_signs = -1
let g:ale_maximum_file_size = v:null
let g:ale_open_list = 0
let g:ale_pattern_options = v:null
let g:ale_pattern_options_enabled = v:null
let g:ale_set_balloons = 0
let g:ale_set_highlights = 1
let g:ale_set_loclist = 1
let g:ale_set_quickfix = 0
let g:ale_set_signs = 1
let g:ale_sign_column_always = 1
let g:ale_sign_error = 'XX'
let g:ale_sign_info = '!!'
let g:ale_sign_offset = 1000000
let g:ale_sign_style_error = 'XX'
let g:ale_sign_style_warning = '!!'
let g:ale_sign_warning = '!!'
let g:ale_sign_highlight_linenrs = 0
let g:ale_statusline_format = v:null
let g:ale_type_map = {}
let g:ale_use_global_executables = v:null
let g:ale_virtualtext_cursor = 0
let g:ale_warn_about_trailing_blank_lines = 1
let g:ale_warn_about_trailing_whitespace = 1
  Command History:

(executable check - success) kotlinc
(finished - exit code 1) ['/bin/zsh', '-c', 'kotlinc    ''/home/lovecraft/hello.kt''']

<<<NO OUTPUT RETURNED>>>

(executable check - failure) ktlint
```

**Comment by ndtho8205** [comment]
I am also dealing with the same problem. Big thanks for your suggestion, it seems like the code forgot to set `output_stream` when running the kotlinc linter ([kotlinc.vim#L176](https://github.com/dense-analysis/ale/blob/8f7ccdc5e95302223c516a0113f5af7e075dc041/ale_linters/kotlin/kotlinc.vim#L176)).

```
'output_stream': 'stderr'
```
So I added one line only and it works.

```
call ale#linter#Define('kotlin', {
\   'name': 'kotlinc',
\   'executable': 'kotlinc',
\   'output_stream': 'stderr',           <------ add this line
\   'command': function('ale_linters#kotlin#kotlinc#RunWithImportPaths'),
\   'callback': 'ale_linters#kotlin#kotlinc#Handle',
\   'lint_file': 1,
\})
```
I also took a look to the test, but really I can't understand how to verify it :))) So I guess this is just a workaround.

**Comment by RyanSquared** [comment]
the tests work based on an example output which is usually copy pasted from a terminal, so this change shouldn't affect tests. if you'd like to open a PR and test it out on some actual kotlin project that would be helpful.

**Comment by ndtho8205** [comment]
I tested it on my kotlin + maven project, so far it works fine. So I make a pull request. Just a really small change :) Btw, my first ever pull request to a project that I really like :))

**Comment by hsanson** [comment]
@ndtho8205 do you mind telling me which version of kotlinc you are using?  trying this PR I cannot get ALE work with Kotlin/Native: 1.3.72.  ALE fails with `invalid argument: -cp`.

Just want to verify if kotlinc outputs all important information only to stderr and not both stderr and stdout before merging this.

**Comment by ndtho8205** [comment]
Sorry for my late response.

There are two types of Kotlin compiler: `kotlinc-jvm` and `kotlinc-native` (you can download them from [here](https://github.com/JetBrains/kotlin/releases/tag/v1.3.72)). The interface of those two are completely different. ALE linter currently only works on `kotlinc-jvm` which have argument -cp (--classpath) to include directories for searching java class files. Since you are using `kotlinc-native`, the ALE will fail.

```
$ kotlinc -help             #<--- the compiler that I installed (kotlin-compiler-1.3.72.zip)
Usage: kotlinc-jvm <options> <source files>
where possible options include:
  -classpath (-cp) <path>    List of directories and JAR/ZIP archives to search for user class files
  -d <directory|jar>         Destination for generated class files
  -include-runtime           Include Kotlin runtime into the resulting JAR
  -java-parameters           Generate metadata for Java 1.8 reflection on method parameters
  ...

$ ./kotlinc -help             #<--- file extracted from kotlin-native-linux-1.3.72.tar.gz
Usage: kotlinc-native <options> <source files>
where possible options include:
  -g                         Enable emitting debug information
  -enable-assertions (-ea)   Enable runtime assertions in generated code
  -friend-modules <path>     Paths to friend modules
  -generate-no-exit-test-runner (-trn)
  ...
```

Regarding the outputs of `kotlinc`, you can check its test cases from [here](https://github.com/JetBrains/kotlin/blob/5ab55fbb57/compiler/testData/cli/jvm/wrongAbiVersion.out) and [here](https://github.com/JetBrains/kotlin/blob/5ab55fbb57/compiler/tests/org/jetbrains/kotlin/cli/LauncherScriptTest.kt). Based on these test cases, I think that all warning and errors will go to stderr.

**Comment by hsanson** [comment]
Thanks, was able to verify the PR works as expected.
